### ISSUE TITLE
Bring Rx module up to RxJava3

### ISSUE THREAD

**Issue by ryanhubbell** [HubbellCorp/SweetBlue#21]
RxJava2 is now deprecated, and only receiving bug fix updates.

**Comment by Whathecode** [comment]
Yes please! And other libraries we use depend on RxJava3, so currently we are forced to pull in both.

**Comment by ryanhubbell** [comment]
Please see https://github.com/HubbellCorp/SweetBlue/discussions/22
